public/app/features/query/components/QueriesTab.tsx

```diff
@@ -24,14 +21,23 @@ import {
 import { PluginHelp } from 'app/core/components/PluginHelp/PluginHelp';
 import { addQuery } from 'app/core/utils/query';
 import { Unsubscribable } from 'rxjs';
-import { DashboardQueryEditor, isSharedDashboardQuery } from 'app/plugins/datasource/dashboard';
 import { expressionDatasource, ExpressionDatasourceID } from 'app/features/expressions/ExpressionDatasource';
-import { css } from 'emotion';
 import { selectors } from '@grafana/e2e-selectors';
+import { PanelQueryRunner } from '../state/PanelQueryRunner';
+import { QueryGroupOptions, QueryGroupOptionsEditor } from './QueryGroupOptions';
+import { DashboardQueryEditor, isSharedDashboardQuery } from 'app/plugins/datasource/dashboard';
+import { css } from 'emotion';
 
 interface Props {
-  panel: PanelModel;
-  dashboard?: DashboardModel;
+  queryRunner: PanelQueryRunner;
+  queries: DataQuery[];
+  dataSourceName: string | null;
+  options: QueryGroupOptions;
+  onOpenQueryInspector?: () => void;
+  onRunQueries: () => void;
+  onQueriesChange: (queries: DataQuery[]) => void;
+  onDataSourceChange: (ds: DataSourceSelectItem, queries: DataQuery[]) => void;
+  onOptionsChange: (options: QueryGroupOptions) => void;
```

Comment:
thinking of moving queries & dataSourceName into the QueryGroupOptions to simplify this interface a bit,

we could then get rid of onDataSourceChange and onQueriesChange as we can cover that with onOptionsChange. Thoughts?

Reply:
ok, going to merge this and do this in a follow-up PR